redwood-treehouse-lazylayout-view/src/main/kotlin/app/cash/redwood/treehouse/lazylayout/view/ViewLazyColumn.kt

```diff
@@ -15,118 +15,157 @@
 import app.cash.redwood.LayoutModifier
 import app.cash.redwood.treehouse.AppService
+import app.cash.redwood.treehouse.Content
+import app.cash.redwood.treehouse.HostConfiguration
 import app.cash.redwood.treehouse.TreehouseApp
 import app.cash.redwood.treehouse.TreehouseContentSource
 import app.cash.redwood.treehouse.TreehouseView.WidgetSystem
 import app.cash.redwood.treehouse.TreehouseWidgetView
-import app.cash.redwood.treehouse.bindWhenReady
 import app.cash.redwood.treehouse.lazylayout.api.LazyListIntervalContent
 import app.cash.redwood.treehouse.lazylayout.widget.LazyColumn
-import okio.Closeable
-
-private data class LazyContentItem(
-  val index: Int,
-  val item: LazyListIntervalContent.Item,
-)
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
 
 internal class ViewLazyColumn<A : AppService>(
-  treehouseApp: TreehouseApp<A>,
+  private val treehouseApp: TreehouseApp<A>,
   widgetSystem: WidgetSystem,
   override val value: RecyclerView,
 ) : LazyColumn<View> {
+  private val scope = MainScope()
+
   override var layoutModifiers: LayoutModifier = LayoutModifier
 
-  private val adapter = LazyContentItemListAdapter(
-    treehouseApp,
-    widgetSystem,
-    contentHeight = TypedValue.applyDimension(
-      TypedValue.COMPLEX_UNIT_DIP,
-      64F,
-      value.resources.displayMetrics,
-    ).toInt(),
-  )
+  private val adapter = LazyContentItemListAdapter(widgetSystem)
 
   init {
     value.apply {
       layoutManager = LinearLayoutManager(value.context)
       layoutParams = ViewGroup.LayoutParams(MATCH_PARENT, WRAP_CONTENT)
     }
     value.adapter = adapter
-  }
+    value.addOnAttachStateChangeListener(
+      object : View.OnAttachStateChangeListener {
+        override fun onViewAttachedToWindow(view: View) {}
 
-  override fun intervals(intervals: List<LazyListIntervalContent>) {
-    adapter.submitList(
-      intervals.flatMap { interval ->
-        List(interval.count) { index ->
-          LazyContentItem(index, interval.itemProvider)
+        override fun onViewDetachedFromWindow(view: View) {
+          view.removeOnAttachStateChangeListener(this)
+          scope.cancel()
         }
       },
     )
   }
 
-  private class LazyContentItemListAdapter<A : AppService>(
-    private val treehouseApp: TreehouseApp<A>,
-    private val widgetSystem: WidgetSystem,
-    private val contentHeight: Int,
-  ) : ListAdapter<LazyContentItem, ViewHolder>(LazyContentItemDiffCallback) {
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-      val container = FrameLayout(parent.context).apply {
-        layoutParams = ViewGroup.LayoutParams(MATCH_PARENT, contentHeight)
+  override fun intervals(intervals: List<LazyListIntervalContent>) {
+    // TODO Don't hardcode pageSizes
+    // TODO Enable placeholder support
+    // TODO Set a maxSize so we don't keep _too_ many views in memory
```

Comment:
I'll do all three of these in follow-ups as they're needed.

I wanted to do the last point in this PR, but I don't think Paging allows you to be notified of a drop event _with_ the page that was dropped. I'm gonna create an Issue with them, otherwise I'll need to figure out a workaround.